tests/code/test_spreadsheet.py

```diff
@@ -0,0 +1,196 @@
+        assert list(repeatVolunteers().execute()) == [('Sreynit Khatt', 4), ('Zach Neill', 2), ('Tyler Parton', 2)]
+        transaction.rollback()
+
+@pytest.mark.integration
+def test_repeatVolunteersPerProgram():
+    with mainDB.atomic() as transaction:
+        assert list(repeatVolunteersPerProgram().execute()) == [('Zach Neill', 'Hunger Initiatives', 2), ('Sreynit Khatt', 'Adopt-a-Grandparent', 3)]
+        EventParticipant.create(user = 'partont',
+                                event = 2,
+                                hoursEarned = 1)
+        assert list(repeatVolunteersPerProgram().execute()) == [('Zach Neill', 'Hunger Initiatives', 2), ('Tyler Parton', 'Hunger Initiatives', 2),
+                                                                ('Sreynit Khatt', 'Adopt-a-Grandparent', 3)]
+
+        transaction.rollback()
+
+@pytest.mark.integration
+def test_volunteerMajorAndClass():
+    with mainDB.atomic() as transaction:
+        assert list(volunteerMajorAndClass(User.major).execute()) == [('Biology', 1), ('Chemistry', 1), ('Computer Science', 2), ('Psychology', 1)]
+        User.create(username = 'solijonovam',
+                    bnumber = 'B00769465',
+                    email = 'solijonovam@berea.edu',
+                    phoneNumber = '732-384-3469',
+                    firstName = 'Madinabonu',
+                    lastName  = 'Solijonova',
+                    isStudent = True,
+                    major = 'Agriculture',
+                    classLevel = 'Sophomore')
+        EventParticipant.create(user = 'solijonovam',
+                                event = 2,
+                                hoursEarned = 2)
+        assert list(volunteerMajorAndClass(User.major).execute()) == [('Agriculture', 1), ('Biology', 1), ('Chemistry', 1), ('Computer Science', 2), ('Psychology', 1)]
+        assert list(volunteerMajorAndClass(User.classLevel).execute()) == [('Junior', 1), ('Senior', 3), ('Sophomore', 2)]
+        assert list(volunteerMajorAndClass(User.classLevel, True).execute()) == [('Sophomore', 2), ('Junior', 1), ('Senior', 3)]
+        transaction.rollback()
+
+@pytest.mark.integration
+def test_volunteerHoursByProgram():
+    with mainDB.atomic() as transaction:
+        assert list(volunteerHoursByProgram().execute()) == [('Adopt-a-Grandparent', 9.0), ('Berea Buddies', 0.0), ('Hunger Initiatives', 11.0)]
+        EventParticipant.create(user = 'partont',
+                                event = 2,
+                                hoursEarned = 1)
+        assert list(volunteerHoursByProgram().execute()) == [('Adopt-a-Grandparent', 9.0), ('Berea Buddies', 0.0), ('Hunger Initiatives', 12.0)]
+        transaction.rollback()
+
+@pytest.mark.integration
+def test_onlyCompletedAllVolunteer():
+    #seems like onlyCompletedAllVolunteer isn't working properly
+    with mainDB.atomic() as transaction:
+        assert list(onlyCompletedAllVolunteer("2020-2021").execute()) == []
```

Comment:
Did you mean to fix it eventually? (comment says it doesn't work properly). Also, this function in particular could use testing invalid inputs.

Reply:
This function has been changed to work as intended, the comment has been removed. There are negative tests in this function as there are people from 2020-2021 who have not completed all volunteer events and are not showing up as intended.